rts/lra/test/arquillian-extension/src/test/java/io/narayana/lra/arquillian/spi/NarayanaLRARecoveryTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2020, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian.spi;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URI;
+
+public class NarayanaLRARecoveryTest {
+
+    @Test
+    public void extractCoordinatorFromUsualLRAId() {
```

Comment:
Why are we testing internals.
If our code is constructing invalid LRA ids then a majority of tests will end up failing.
These tests are just adding noise to our code base.

Reply:
I don't think so. This is a unit test which is normal to be used to check the functionality of particular parts of a class. The tests help to do changes and refactoring and not breaking the functionality.

And then there are the integration tests that check if the whole machinery works together.

Reply:
Unit tests are the first step of testing, I agree.

Reply:
After thinking about this I would like to preserve the unit test. I'm in agreement with Mayank here.